Packs/AWS_SystemManager/Integrations/AWSSystemManager/AWSSystemManager.py

```diff
@@ -0,0 +1,647 @@
+
+
+def next_token_command_result(next_token: str, outputs_prefix: str) -> CommandResults:
+    """Creates a CommandResults object with the next token as the output.
+
+    Args:
+    ----
+        next_token (str): The next token.
+        outputs_prefix (str): The prefix for the outputs.
+
+    Returns:
+    -------
+        CommandResults: A CommandResults object with the next token as the output.
+    """
+    return CommandResults(
+        outputs_prefix=f"AWS.SSM.{outputs_prefix}",
+        outputs=next_token,
+        readable_output=f"For more results rerun the command with {next_token=}.",
+    )
+
+
+""" COMMAND FUNCTIONS """
+
+
+def add_tags_to_resource_command(ssm_client, args: dict[str, Any]) -> CommandResults:
+    """Adds tags to a specified resource.
+    The response from the API call when success is empty dict.
+
+    Args:
+    ----
+        ssm_client (SSMClient): An instance of the SSM client.
+        args (dict): A dictionary containing the command arguments.
+                     - 'resource_type' (str): The type of the resource.
+                     - 'resource_id' (str): The ID of the resource.
+                     - 'tag_key' (str): The key of the tag to add.
+                     - 'tag_value' (str): The value of the tag to add.
+
+    Returns:
+    -------
+        CommandResults: readable output only,
+    """
+    kwargs = {
+        "ResourceType": args["resource_type"],
+        "ResourceId": args["resource_id"],
+        "Tags": [{"Key": args["tag_key"], "Value": args["tag_value"]}],
+    }
+
+    ssm_client.add_tags_to_resource(**kwargs)
+    return CommandResults(
+        readable_output=f"Tags added to resource {args['resource_id']} successfully.",
+    )
+
+
+def get_inventory_command(ssm_client, args: dict[str, Any]) -> list[CommandResults]:
+    """Fetches inventory information from AWS SSM using the provided SSM client and arguments.
+
+    Args:
+    ----
+        ssm_client: SSM client object for making API requests.
+        args (dict): Command arguments containing filters and parameters.
+
+    Returns:
+    -------
+        list[CommandResults]: A list of CommandResults containing the inventory information.
+    """
+    def _parse_inventory_entities(entities: list[dict]) -> list[dict]:
+        """Parses a list of entities and returns a list of dictionaries containing relevant information.
+
+        Args:
+        ----
+            entities: A list of entities to parse.
+
+        Returns:
+        -------
+            list of dict containing relevant information.
+        """
+        parsed_entities = []
+        for entity in entities:
+            entity_content = dict_safe_get(entity, ["Data", "AWS:InstanceInformation", "Content"], [{}])
+            parsed_entity = {"Id": entity.get("Id")}
+            for content in entity_content:
+                parsed_entity.update(
+                    {
+                        "Instance Id": content.get("InstanceId"),
```

Comment:
what if the first entity has this field, then the second one doesn't? Do we overwrite it with `None`?